lib/levenshtein.nix

```diff
@@ -0,0 +1,132 @@
+{ lib
+}:
+
+/*
+  Created by infinisil for https://github.com/NixOS/nixpkgs/pull/79877
+  SPDX-License-Identifier: MIT
+*/
+
+rec {
+  /* Computes the Levenshtein distance between two strings.
+     Complexity O(n*m) where n and m are the lengths of the strings.
+     Algorithm adjusted from https://stackoverflow.com/a/9750974/6605742
+     Type: levenshtein :: string -> string -> int
+     Example:
+       levenshtein "foo" "foo"
+       => 0
+       levenshtein "book" "hook"
+       => 1
+       levenshtein "hello" "Heyo"
+       => 3
+  */
+  levenshtein = a: b: let
```

Comment:
I kept the original formatting to make copying easier. Ideally this would be in a repo we could just include.